src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java

```diff
@@ -140,4 +142,117 @@ public abstract void doFilter (HttpExchange exchange, Chain chain)
      */
     public abstract String description ();
 
+    /**
+     * Returns a pre-processing {@code Filter} with the given description and
+     * operation.
+     *
+     * <p>The {@code description} string describes the returned filter. The
+     * {@link Consumer operation} is the effective implementation of the filter.
+     * It is executed for each {@code HttpExchange} before invoking either the
+     * next filter in the chain or the exchange handler (if this is the final
+     * filter in the chain).
+     *
+     * @apiNote
+     * A beforeHandler filter is typically used to examine or modify the
+     * exchange state before it is handled. The filter operation is executed
+     * before {@link Filter.Chain#doFilter(HttpExchange)} is invoked, so before
+     * any subsequent filters in the chain and the exchange handler are
+     * executed. The response is commonly sent by the exchange handler.
+     * The filter is only expected to send the response in the uncommon case
+     * that the exchange will not be handled after the filter is executed.
```

Comment:
Maybe this could be rephrased as: 

```
 * The filter {@code operation} is typically not expected to handle the request or 
 * {@linkplain HttpExchange#sendResponseHeaders(int, long) send response headers}, since
 * the next filter in the chain (or the exchange handler) will eventually be invoked after the 
 * operation completes.
```

Alternatively we could use a `Predicate<HttpExchange>` instead of a consumer to tell whether the next filter in the chain should be invoked - but that would be another kind of worms. I believe `Consumer<HttpExchange>` is the API we want.